File: alura/php/loja/classe/Produto.php
<?php
  class Produto {
    public $id;
    public $nome;
    public $preco;
    public $descricao;
    public $categoria;
    public $usado;

    public function valorComDesconto($valor = 0.05) {
      if ($valor > 0 && $valor < 1) {
        $this->preco -= $this->preco * $valor;
        return $this->preco;
      }
    }
  }
?>

File: alura/algoritmos/Produto.java
public class Produto {

  private String nome;
  private double preco;

  public Produto(String nome, double preco) {
    this.nome = nome;
    this.preco = preco;
  }

  public double getPreco() {
    return this.preco;
  }

  public String getNome() {
    return this.nome;
  }

}
